tests/test_model_builder.py

```diff
@@ -0,0 +1,294 @@
+DMXF2_0002        1442.000
+FD1  1.61666384D-04  1      3.38650356D-05
+FD2 -1.88210030D-04  1      4.13173074D-05
+SOLARN0               0.00
+EPHEM               DE421
+ECL                 IERS2003
+CLK                 TT(BIPM)
+UNITS               TDB
+TIMEEPH             FB90
+T2CMETHOD           TEMPO
+CORRECT_TROPOSPHERE N
+PLANET_SHAPIRO      N
+#DILATEFREQ          N
+NTOA                  4005
+TRES                  5.52
+TZRMJD  54981.28084616488447
+TZRFRQ             424.000
+TZRSITE                  3
+NITS                     1
+INFO -f
+BINARY            DD
+A1             9.230780480  1         0.000000203
+E             0.0000216340  1        0.0000000236
+T0        54975.5128660817  1        0.0019286695
+PB       12.32717119132762  1    0.00000000019722
+OM        276.536118059963  1      0.056323656112
+SINI              0.999461  1            0.000178
+M2                0.233837  1            0.011278
+RNAMP         0.17173D-01
+RNIDX            -4.91353
+TNRedAmp -14.227505410948254
+TNRedGam 4.91353
+TNRedC 45
+T2EFAC -f L-wide_PUPPI   1.507
+T2EFAC -f 430_ASP   1.147
+T2EQUAD -f L-wide_PUPPI   0.25518
+T2EQUAD -f 430_ASP   0.01410
+ECORR -f 430_PUPPI   0.00601
+ECORR -f L-wide_PUPPI   0.31843
+JUMP -fe L-wide      -0.000009449  1       0.000009439
+"""
+
+
+class SimpleModel(PhaseComponent):
+    """Very simple test model component
+    """
+
+    register = True
+    category = "simple_test"
+
+    def __init__(self):
+        super(SimpleModel, self).__init__()
+        self.add_param(floatParameter(name="TESTPARAM", value=0.0, unit="s"))
+
+
+class SubsetModel(PhaseComponent):
+    """Test model component hosting the parameters which are a subset of spindown.
+    """
+
+    register = False  # This has to be false, otherwrise all test will fail.
+    category = "simple_test"
+
+    def __init__(self):
+        super(SubsetModel, self).__init__()
+        self.add_param(floatParameter(name="F0", value=0.0, unit="1/s"))
+        self.add_param(floatParameter(name="F1", value=0.0, unit="1/s^2"))
+
+
+@pytest.fixture
+def sub_set_model():
+    return SubsetModel()
+
+
+@pytest.fixture
+def simple_model():
+    return SimpleModel()
+
+
+@pytest.fixture
+def simple_model_overlap(simple_model):
+    simple_model.add_param(floatParameter(name="F0", aliases=[], value=0.0, unit="1/s"))
+    return simple_model
+
+
+@pytest.fixture
+def simple_model_alias_overlap():
+    simple_model = SimpleModel()
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM2", aliases=["F0"], value=0.0, unit="1/s")
+    )
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM3", aliases=["LAMBDA"], value=0.0, unit="deg")
+    )
+    return simple_model
+
+
+def test_model_builder_class():
+    """Test if model builder collected components information correctly
+    """
+    mb = ModelBuilder()
+    category = mb.category_component_map
+    assert len(mb.param_component_map["PX"]) == len(category["astrometry"])
```

Comment:
This will break if someone adds an astrometry model that doesn't include PX, and that breakage will not indicate that there's anything wrong with ModelBuilder.